drivers/ec_host_cmd_periph/ec_host_cmd_periph_shi_ite.c

```diff
@@ -0,0 +1,504 @@
+{
+	struct shi_it8xxx2_data *data = CONTAINER_OF(cb, struct shi_it8xxx2_data, cs_cb);
+
+	if (data->shi_state == SHI_STATE_DISABLED) {
+		return;
+	}
+
+	/* Move to processing state */
+	shi_ite_set_state(data, SHI_STATE_PROCESSING);
+}
+
+static int shi_ite_init_registers(const struct device *dev)
+{
+	const struct shi_it8xxx2_cfg *const cfg = dev->config;
+	/* Set FIFO data target count */
+	struct ec_host_cmd_request_header cmd_head;
+	int status;
+
+	/*
+	 * Target count means the size of host request.
+	 * And plus extra 4 bytes because the CPU accesses FIFO base on word. If host requested
+	 * data length is one byte, we need to align the data length to 4 bytes.
+	 */
+	int target_count = sizeof(cmd_head) + 4;
+	/* Offset of data_len member of host request. */
+	int offset = (char *)&cmd_head.data_len - (char *)&cmd_head;
+
+	IT83XX_SPI_FTCB1R = (target_count >> 8) & 0xff;
+	IT83XX_SPI_FTCB0R = target_count & 0xff;
+
+	/*
+	 * The register setting can capture the length field of host
+	 * request.
+	 */
+	IT83XX_SPI_TCCB1 = (offset >> 8) & 0xff;
+	IT83XX_SPI_TCCB0 = offset & 0xff;
+
+	/*
+	 * Memory controller configuration register 3.
+	 * bit6 : SPI pin function select (0b:Enable, 1b:Mask)
+	 */
+	IT83XX_GCTRL_MCCR3 |= IT83XX_GCTRL_SPISLVPFE;
+
+	/* Set unused blocked byte */
+	IT83XX_SPI_HPR2 = 0x00;
+
+	/* Rx valid length interrupt enabled */
+	IT83XX_SPI_RX_VLISMR &= ~IT83XX_SPI_RVLIM;
+
+	/*
+	 * General control register2
+	 * bit4 : Rx FIFO2 will not be overwrited once it's full.
+	 * bit3 : Rx FIFO1 will not be overwrited once it's full.
+	 * bit0 : Rx FIFO1/FIFO2 will reset after each CS_N goes high.
+	 */
+	IT83XX_SPI_GCR2 = IT83XX_SPI_RXF2OC | IT83XX_SPI_RXF1OC | IT83XX_SPI_RXFAR;
+
+	/*
+	 * Interrupt mask register (0b:Enable, 1b:Mask)
+	 * bit5 : Rx byte reach interrupt mask
+	 * bit2 : SPI end detection interrupt mask
+	 */
+	IT83XX_SPI_IMR &= ~IT83XX_SPI_EDIM;
+
+	/* Reset fifo and prepare to for next transaction */
+	shi_ite_reset_rx_fifo();
+
+	/* Ready to receive */
+	shi_ite_set_state(dev->data, SHI_STATE_READY_TO_RECV);
+
+	/* Interrupt status register(write one to clear) */
+	IT83XX_SPI_ISR = 0xff;
+
+	/* SPI peripheral controller enable (after settings are ready) */
+	IT83XX_SPI_SPISGCR = IT83XX_SPI_SPISCEN;
+
+	/* Set the pin to SHI alternate function. */
+	status = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
+	if (status < 0) {
+		LOG_ERR("Failed to configure SHI pins");
+		return status;
+	}
+
+	/* Enable SPI peripheral interrupt */
+	IRQ_CONNECT(DT_INST_IRQN(0), DT_INST_IRQ(0, priority), shi_ite_int_handler,
+		    DEVICE_DT_INST_GET(0), 0);
+	irq_enable(DT_INST_IRQN(0));
+
+	/* Configure the SPI chip select */
+	status = gpio_pin_configure(cfg->cs.port, cfg->cs.pin, GPIO_INPUT | cfg->cs.dt_flags);
+	if (status < 0) {
+		LOG_ERR("Failed to configure SHI CS pin");
+		return status;
+	}
+
+	/* Enable SPI chip select pin interrupt */
+	status = gpio_pin_interrupt_configure(cfg->cs.port, cfg->cs.pin,
```

Comment:
The gpio interrupt enable here is duplicated declaration as line 440.
The line 440 declaration should be reserved and the flag set to edge falling is right.

Reply:
Thanks for catching this, removed the redundant configuration.